### PR TITLE
Expose more information to python functions

### PR DESCRIPTION
**Pull request by dengdifan**
Additional RF components (trees, nodes, and splits) are now exposed to the python functions and thus more interactions are allowed

### CODE REVIEW

File: include/rfr/nodes/k_ary_node.hpp [hunk 1]
```diff
@@ -31,6 +31,11 @@ template <int k, typename split_type, typename num_t = float, typename response_
 class k_ary_node_minimal{
   protected:
 	index_t parent_index;
+	index_t depth; // current depth of the node
+
+	index_t num_data; // number of data points stored in this node
+	//TODO do we need to store all the data indices in all nodes or only in leaf nodes
```
Comment (AndreBiedenkapp) [inline include/rfr/nodes/k_ary_node.hpp:37]: Is this TODO resolved by now?

File: include/rfr/nodes/k_ary_node.hpp [hunk 2]
```diff
@@ -283,8 +311,43 @@ class k_ary_node_full: public k_ary_node_minimal<k, split_type, num_t, response_
...
+
+    /** \brief get the number of data points stored in the tree derived from this node*/
+    virtual index_t get_num_data() const{return super::num_data;}
+
 };
 
 
 }} // namespace rfr::nodes
-#endif
+#endif":wq"
```
Comment (AndreBiedenkapp) [inline include/rfr/nodes/k_ary_node.hpp:353]: Could you please remove this artifact :)

**Review by AndreBiedenkapp — CHANGES_REQUESTED** [review]
Sorry for the late review. I only got notified once @mfeurer assigned me as reviewer.
There are only two minor changes that need to be done before we can merge this PR.

**Comment by dengdifan** [comment]
I updated the PR with your comment, I'll later add some tests accordingly

**Pull request merged**